src/controllers/user.controller.ts

```diff
@@ -13,44 +12,47 @@ class UserController {
     this.userService = new UserService();
   }
 
-  getAll = async (req: Request, res: Response) => {
+  async getAll(request: Request, response: Response): Promise<void> {
     try {
-      const users: UserResponse[] = await this.userService.findAll();
-      res.status(200).json({
+      const users = await this.userService.findAll();
+      
+      response.status(200).json({
         success: true,
         message: 'User retrieved successfully.',
         data: users,
       });
     } catch (error) {
-      const e = error as Error;
-      res.status(500).json({
+      response.status(500).json({
         success: false,
         message: 'An unexpected error occurs',
       });
 
-      logger.error(e.message);
+      logger.error((error as Error).message);
     }
   };
 
-  getById = async (request: Request, response: Response) => {
+  async getById(request: Request, response: Response): Promise<void> {
     const id: string = request.params.id;
+    
     try {
-      const user: UserResponse | null = await this.userService.findById(id);
-
-      if (user) {
-        response.status(200).send({
-          success: true,
-          status: 200,
-          message: 'User retrieved successfully',
-          data: user,
-        });
-      } else {
+      const user = await this.userService.findById(id);
```

Comment:
user type has been omitted

Reply:
The type is already perfectly infered from `userService.findById`

![Screenshot from 2023-07-28 12-16-51](https://github.com/eliotfgn/talkative-be/assets/64146788/2b205cb5-7a6d-4816-8877-42cff86c05e9)

Typing `user` manually is not recommended as it can lead to uncaught type errors if badly done.